src/extras/amd.js

```diff
@@ -88,6 +88,10 @@
     if (register && register[1] === lastRegisterDeclare)
       return register;
 
+    // If the script registered a named module, return that module instead of re-instantiating it.
+    if (firstNamedDefine)
+      return firstNamedDefine;
```

Comment:
Before this PR, we were calling `createAMDRegister()` multiple times for the same AMD module. This is problematic because (1) some modules rely on their code only being executed once,  and (2) performance of running the execute function multiple times.